client/hotrod-client/src/test/java/org/infinispan/client/hotrod/near/NearCacheMarshallingTest.java

```diff
@@ -0,0 +1,96 @@
+package org.infinispan.client.hotrod.near;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.configuration.NearCacheMode;
+import org.infinispan.client.hotrod.test.SingleHotRodServerTest;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.jboss.marshalling.commons.GenericJBossMarshaller;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "client.hotrod.near.NearCacheMarshallingTest")
+public class NearCacheMarshallingTest extends SingleHotRodServerTest {
+
+   private final Class<? extends Marshaller> marshaller;
+   private final MediaType storeType;
+
+   protected NearCacheMarshallingTest(Class<? extends Marshaller> marshaller, MediaType storeType) {
+      this.marshaller = marshaller;
+      this.storeType = storeType;
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      org.infinispan.configuration.cache.ConfigurationBuilder serverCfg = new org.infinispan.configuration.cache.ConfigurationBuilder();
+      if (storeType != null) hotRodCacheConfiguration(serverCfg, storeType);
+      return TestCacheManagerFactory.createCacheManager(contextInitializer(), serverCfg);
+   }
+
+   @Override
+   protected void setup() throws Exception {
+      cacheManager = createCacheManager();
+      hotrodServer = createHotRodServer();
+   }
+
+   @Override
+   protected RemoteCacheManager getRemoteCacheManager() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addServer().host("127.0.0.1").port(hotrodServer.getPort());
+      if (marshaller != null) builder.marshaller(marshaller);
+      builder.remoteCache("").nearCacheMode(NearCacheMode.INVALIDATED)
+            .nearCacheMaxEntries(2)
+            // Bloom filter needs to be enabled.
+            .nearCacheUseBloomFilter(true);
```

Comment:
Can we add using bloom filter or not to the factory method. This way we are sure we cover the non bloom filter case as well.

Reply:
Updated.